storage/innobase/buf/buf0flu.cc

```diff
@@ -2095,9 +2120,11 @@ static os_thread_ret_t DECLARE_THREAD(buf_flush_page_cleaner)(void*)
     else if (srv_shutdown_state > SRV_SHUTDOWN_INITIATED)
       break;
 
-    if (buf_pool.page_cleaner_idle())
-      my_cond_wait(&buf_pool.do_flush_list,
-                   &buf_pool.flush_list_mutex.m_mutex);
+    if (buf_pool.page_cleaner_idle()
+	&& ((srv_max_dirty_pages_pct_lwm != 0
+             && UT_LIST_GET_LEN(buf_pool.flush_list) == 0)
+            || (srv_max_dirty_pages_pct_lwm == 0)))
+      my_cond_wait(&buf_pool.do_flush_list, &buf_pool.flush_list_mutex.m_mutex);
```

Comment:
I think that your expression could be simpilfied to the following:
```c++

    if (buf_pool.page_cleaner_idle() &&
        (!srv_max_dirty_pages_pct_lwm ||
         !UT_LIST_GET_LEN(buf_pool.flush_list)))
```
(Note: We do not follow Heikki Tuuri’s style of splitting a line before an operator, but the common style of splitting after an operator.)
This condition may be causing unnecessary wake-ups in the case that `innodb_max_dirty_pages_pct_lwm` has been set but the `dirty_pct` does not exceed it. But, it would not be easy to move the `dirty_pct` computation earlier. To compensate for this, I think that we can add a `my_cond_wait()` call before the `continue` in the following:
```c++
unemployed:
      buf_pool.page_cleaner_set_idle(true);
      continue;
```

Reply:
I verified that my suggestion to add the `my_cond_wait()` call does not work.